### PR TITLE
Move flutter tests from packages/unit to packages/flutter/test

### LINKED ISSUE #50: Fold packages/unit into packages/flutter
There's no reason to keep these separate anymore.  We can just have `packages/unit/test` be `packages/flutter/test` and make `test` and `quiver` dev dependencies of the flutter package.

### PR DESCRIPTION
**Pull request by Hixie**
Fixes #50.

### CODE REVIEW

File: packages/flutter/pubspec.yaml [hunk 1]
```diff
@@ -19,5 +22,13 @@ dependencies:
   newton:
     path: ../newton
 
+dev_dependencies:
+  flx:
+    path: ../flx
```
Comment (abarth) [inline packages/flutter/pubspec.yaml]: Why not move the flx tests into package/flx/test ?

File: packages/flutter/pubspec.yaml [hunk 2]
```diff
@@ -19,5 +22,13 @@ dependencies:
...
+  flutter_test:
+    path: ../flutter_test
+  stocks:
+    path: ../../examples/stocks
```
Comment (abarth) [inline packages/flutter/pubspec.yaml]: Why not move the stocks tests into the stocks app?  This dependency from production code to example code is backwards.

File: packages/flutter/pubspec.yaml (hunk 1 above)
Comment (Hixie) [inline packages/flutter/pubspec.yaml]: I hadn't even thought of it. Will do.

Comment (Hixie) [inline packages/flutter/pubspec.yaml]: Looks like they're already there. Maybe we don't need this dependency? This was just me merging in the dependencies we had for the unit/ directory.

File: packages/flutter/pubspec.yaml (hunk 2 above)
Comment (Hixie) [inline packages/flutter/pubspec.yaml]: yup, will do. Sorry, I hadn't really finished this yet, I was mostly testing to see what Travis would do. :-)

**Comment by Hixie** [comment]
There's really not much point having `flutter test --flutter-repo` any more other than our own convenience, is there? Shouldn't just running `flutter test` in the `flutter` directory do the same thing?

File: packages/flutter/pubspec.yaml (hunk 2 above)
Comment (Hixie) [inline packages/flutter/pubspec.yaml]: Turns out there's no stocks tests here anyway.

**Comment by Hixie** [comment]
cc @chinmaygarde: This changes the pubspec.yaml files for cassowary and newton. I couldn't work out why you depended on test_runner and dart_coveralls, is it ok to remove them?

**Comment by Hixie** [comment]
@abarth Ok, it's ready for review now. Thanks!

**Comment by chinmaygarde** [comment]
@Hixie I remember I had to do that to get automatic code coverage reports sent to coveralls.io. Also, the default test hook on Travis used test_runner. We don't need those dependencies anymore though.

**Comment by abarth** [comment]
LGTM

We should check CONTRIBUTING.md to see if it refers to packages/unit anymore.

**Comment by Hixie** [comment]
patch now has updated contributor doc

**Pull request merged**